pex/tools/commands/venv.py

```diff
@@ -99,25 +119,83 @@ def record_provenance(src_to_dst):
         for src, dst in src_to_dst:
             provenance[dst].append(src)
 
+    # Since the pex.path() is ~always outside our control (outside ~/.pex), we copy all PEX user
+    # sources into the venv.
     pex_info = pex.pex_info()
     record_provenance(
         _copytree(
             src=PEXEnvironment.mount(pex.path()).path,
             dst=venv.site_packages_dir,
             exclude=(pex_info.internal_cache, pex_info.bootstrap, "__main__.py", pex_info.PATH),
+            symlink=False,
         )
     )
 
     with open(os.path.join(venv.venv_dir, pex_info.PATH), "w") as fp:
         fp.write(pex_info.dump())
 
+    # Since the pex distributions are all materialized to ~/.pex/installed_wheels, which we control,
+    # we can optionally symlink to take advantage of sharing generated *.pyc files for auto-venvs
+    # created in ~/.pex/venvs.
```

Comment:
These symlinks are created entirely within `~/.pex`, right? That seems safe.

Will this impact `pex-tools venv` creating a venv in a user specified location? In that case it might be worth considering whether hardlinks still make sense.

Reply:
If you track callers, the `pex-tools` venv does still use hardlinks/copies.